src/subRecordPicker.js

```diff
@@ -0,0 +1,23 @@
+import createSruClient from '@natlibfi/sru-client';
+import {MARCXML} from '@natlibfi/marc-record-serializers';
+import {createLogger} from './utils';
+
+export function createSubrecordPicker(sruUrl) {
+  const logger = createLogger();
+  const sruClient = createSruClient({url: sruUrl, recordSchema: 'marcxml'});
+
+  return {readSubrecords};
+
+  function readSubrecords(recordId) {
+    logger.log('verbose', `Picking subrecords for ${recordId}`);
+    return new Promise((resolve, reject) => {
+      const records = [];
+      sruClient.searchRetrieve(`melinda.partsofhost=${recordId}`)
+        .on('record', xmlString => {
+          records.push(MARCXML.from(xmlString)); // eslint-disable-line functional/immutable-data
+        })
+        .on('end', () => resolve(records))
+        .on('error', err => reject(err));
+    });
```

Comment:
Use the option `retrieveAll` set to `false` to allow the clients to decide when to request for more subrecords.